pennylane/measurements/__init__.py

```diff
@@ -156,6 +156,38 @@ def circuit(x):
     In PennyLane we use functions to define measurements (e.g. :func:`counts`). These
     functions will return an instance of the corresponding measurement process
     (e.g. :class:`CountsMP`). This decision is just for design purposes.
+
+Adding your new measurement to PennyLane
+----------------------------------------
+
+If you want PennyLane to natively support your new measurement, you have to make a Pull Request
+creating a file with the name of your measurement (e.g. ``state.py``) and add it in
+``pennylane/measurements/``. This file should contain:
+
+* The measurement class with the appropriate ``process`` method defined.
+
+* A function with the same name as the created file that will be used to instantiate the
+measurement class.
+
+.. note:
+
+  Please check the file ``pennylane/measurements/state.py`` for an example.
+
+
+The tests are added to a file of a similar name and location in ``tests/measurements/``.
+
+The class and the function need to be imported in ``pennylane/measurements/__init__.py``. The
+function needs to be imported in ``pennylane/__init__.py``.
+
+Here are a few more tips for adding measurements:
+
+* *Choose the name carefully.* Good names tell the user what the measurement is used for,
+  or what architecture it implements. Ask yourself if a measurement of a similar name could
+  be added soon in a different context.
+
+* *Write good docstrings.* Explain what your measurement does in a clear docstring with ample examples.
+  You find more about Pennylane standards in the guidelines on :doc:`/development/guide/documentation`.
```

Comment:
I like this, I think we also have a webpage with tips for contributors that would be nice to link here. https://docs.pennylane.ai/en/stable/development/guide/documentation.html

Reply:
Added link!